server/src/main/java/org/elasticsearch/cluster/coordination/UnsafeBootstrapMasterCommand.java

```diff
@@ -0,0 +1,187 @@
+            "If you have a backup, restore from the backup instead.\n" +
+            "Running this tool should be the last resort.\n" +
+            "Running this tool may cause arbitrary data loss and may render your cluster completely non functional.\n" +
+            "Do you accept this risk?\n";
+    static final String ABORTED_BY_USER_MSG = "aborted by user";
+    static final String NOT_MASTER_NODE_MSG = "unsafe-bootstrap tool can only be run on master eligible node";
+    static final String FAILED_TO_OBTAIN_NODE_LOCK_MSG = "failed to lock node's directory, is Elasticsearch still running?";
+    static final String NO_NODE_FOLDER_FOUND_MSG = "no node folder is found in data folder(s), node has not been started yet?";
+    static final String NO_NODE_METADATA_FOUND_MSG = "no node meta data is found, node has not been started yet?";
+    static final String NO_MANIFEST_FILE_FOUND_MSG = "no manifest file is found, do you run pre 7.0 Elasticsearch?";
+    static final String GLOBAL_GENERATION_MISSING_MSG = "no metadata is referenced from the manifest file, cluster has never been " +
+            "bootstrapped?";
+    static final String NO_GLOBAL_METADATA_MSG = "failed to find global metadata, metadata corrupted?";
+    static final String EMPTY_LAST_COMMITTED_VOTING_CONFIG_MSG =
+            "last committed voting voting configuration is empty, cluster has never been bootstrapped?";
+    static final String WRITE_METADATA_EXCEPTION_MSG = "exception occurred when writing new metadata to disk";
+    static final String MASTER_NODE_BOOTSTRAPPED_MSG = "Master node was successfully bootstrapped";
+    static final Setting<String> UNSAFE_BOOTSTRAP =
+            ClusterService.USER_DEFINED_META_DATA.getConcreteSetting("cluster.metadata.unsafe-bootstrap");
+
+    UnsafeBootstrapMasterCommand() {
+        super("Unsafely bootstraps the master node if the majority of master eligible nodes is lost");
+        namedXContentRegistry = new NamedXContentRegistry(ClusterModule.getNamedXWriteables());
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        // We don't use classical "private static final Logger", because log4j2 initialization happens
+        // after UnsafeBootstrapMasterCommand instance creation.
+        final Logger logger = LogManager.getLogger(UnsafeBootstrapMasterCommand.class);
+        showWarning(terminal);
+        String text = terminal.readText("Confirm [y/N] ");
+        if (text.equalsIgnoreCase("y") == false) {
+            throw new ElasticsearchException(ABORTED_BY_USER_MSG);
+        }
+
+        Settings settings = env.settings();
+        terminal.println(Terminal.Verbosity.VERBOSE, "Checking node.master setting");
+        Boolean master = Node.NODE_MASTER_SETTING.get(settings);
+        if (master == false) {
+            throw new ElasticsearchException(NOT_MASTER_NODE_MSG);
+        }
+        final int nodeOrdinal = 0;
```

Comment:
I think that we need to be able to deal with higher node ordinals somehow. To be discussed, as it's not immediately clear how to do so.

Reply:
We discussed and concluded there is no good way to do this, but also no real need to support higher node ordinals.